Os/Posix/test/ut/PosixMutexTests.cpp

```diff
@@ -0,0 +1,66 @@
+// ======================================================================
+// \title Os/Posix/test/ut/PosixMutexTests.cpp
+// \brief tests for posix implementation for Os::Mutex
+// ======================================================================
+#include "Os/test/ut/mutex/RulesHeaders.hpp"
+#include "Os/test/ut/mutex/CommonTests.hpp"
+#include "Os/Posix/Task.hpp"
+#include <gtest/gtest.h>
+#include "STest/Scenario/Scenario.hpp"
+#include "Fw/Types/String.hpp"
+
+
+static void testTaskRoutine(void* pointer) {
+    // Os::Test::Mutex::Tester::LockMutex lock_rule;
+    // Os::Test::Mutex::Tester::UnlockMutex unlock_rule;
+
+    Os::Test::Mutex::Tester* tester = reinterpret_cast<Os::Test::Mutex::Tester*>(pointer);
+    
+    for (FwIndexType i = 0; i < 100000; i++) {
+        tester->m_mutex.lock();
+        tester->m_state = Os::Test::Mutex::Tester::MutexState::LOCKED;
+
+        tester->m_value = 1;
+        ASSERT_EQ(tester->m_value, 1);
+
+        tester->m_state = Os::Test::Mutex::Tester::MutexState::UNLOCKED;
+        tester->m_mutex.unLock();
+    }
+
+}
+
+// ----------------------------------------------------------------------
+// Posix Test Cases
+// ----------------------------------------------------------------------
+
+// Attempt to delete a locked mutex - expect an assertion
+TEST_F(FunctionalityTester, PosixDeleteLockedMutex) {
+    Os::Test::Mutex::Tester::LockMutex lock_rule;
+    Os::Test::Mutex::Tester::UnlockMutex unlock_rule;
+    lock_rule.apply(*tester);
+    ASSERT_DEATH_IF_SUPPORTED(delete tester.release(), Os::Test::Mutex::Tester::ASSERT_IN_MUTEX_CPP);
```

Comment:
Why `tester.release()` and not just tester ?

Reply:
`tester` is a `std::unique_ptr`. The `.release()` call here is that of the unique_ptr, **not** Mutex::release(). I will add a comment.

Reply:
But actually, since the deletion is not expected to succeed... maybe we should `get()` instead so that the destructor can be called again and the unique_ptr and underlying Mutex does get destroyed successfully?